bite-size-notes/week-9-d-CKF.md

```diff
@@ -0,0 +1,136 @@
+$$\\dot{x} \\approx \\frac{x\_{t+1}-x\_{t}}{\\Delta t}$$
+where $Δt → 0$.
+
+Substituting this into continuous dynamics, we get,
+
+$$\\frac{x\_{t+1}-x_t}{\\Delta t} = Ax\_{t} + Bu\_{t} + w$$
+$$x\_{t+1} = (I+A \\Delta t)x\_{t} + B \\Delta t u\_{t} + \tilde{w}  \\quad \\quad \\tilde{w} \\sim N(0,Q\\Delta t)$$
+Also,
+$$y\_{t} = Cx\_{t} + \\tilde{v} \\quad \\tilde{v} \\sim N(0,\\frac{R}{\\Delta t})$$
+
+The noise $w(t)$ is integrated over $Δt$ The resulting covariance of
+$w$ after integration is as follows:
+
+$$E[\int \int_{t}^{t+\Delta t} w(\tau)w(\tau ')^T \; d\tau ' d\tau]$$
+$$ = Q(t) \Delta t$$
+
+Similarly, integrating measurement noise over the time step,
+$$\\tilde{v}\_{t} = \\frac{1}{\\Delta t} \\int\_{t}^{t+\\Delta t}v(\\tau)d\\tau$$
+$$E\[\\tilde{V}\_{t} \\tilde{V}\_{t}^T\]=\\frac{R(t)}{\\Delta t}=\\frac{R}{\\Delta t}\\ \\text{(as it is not time-varying)}$$
+With this the continuous time dynamics are turned to discrete time
+dynamics:
+$$x\_{t+1}=(I+A\\Delta t)x\_{t} + B\\Delta t u\_{t} +\\tilde{w}\_{t} \\quad \\quad \\quad \\tilde{w} \\sim \\mathcal{N}(0, Q\\Delta t)$$
+$$y\_{t} = Cx\_{t} + \\tilde{v}\_{t} \\quad \\quad \\quad \\quad \\quad \\quad \\quad \\quad \\quad \\quad \\quad \\quad \\tilde{v} \\sim \\mathcal{N}(0, \\frac{R}{dt})$$
+
+Just like for discrete systems, we have the predict equations given as:
+$$\mu_{t}^{p}=(I+A\Delta t)\mu_{t-1} + B\Delta t u_{t-1}$$
+$$\Sigma_{t}^{p}=(I+A\Delta t)\Sigma_{t-1}(I+A\Delta t)^{T}+Q\Delta t$$
+
+Update equations are as follows:
+$$K\_{t}=\\Sigma\_{t}^p C^{T} (C \\Sigma\_{t}^{p} C^{T} + \\frac{1}{\\Delta t}R)^{-1}$$
+$$\mu_{t}=\mu_{t}^{p}+K_{t}(y_{t}-C\mu_{t}^{p})$$
+$$\Sigma_{t}=(I-K_{t}C)\Sigma_{t}^{p}$$
+
+Rearranging the gain equation we get,
+$$\\frac{1}{\\Delta t}K\_{t} = \\Sigma\_{t}^{p} C^{T} (C \\Sigma\_{t}^{p} C^{T} \\Delta t +R)^{-1}$$
+
+as $\Delta t$ goes to zero, the term on right goes to a finite value of $\Sigma_{t}^{p} C^{T} R^{-1}$. The only way, left hand side can be finite is when $K_t$ also goes to zero.
+This suggests that when continuous-time dynamics are approximated by
+discrete-time dynamics using a small sampling time, the Kalman gain
+approaches zero. Consequently, employing a small sampling time in
+discrete-time dynamics may not be viable.
+
+### Finite difference version of $Σ$
+
+Substituting the update equations into the predict equations for *Σ*, we
+get,
+$$\\Sigma\_{t}^{p}=(I+A\\Delta t)(I-K\_{t-1}C)\\Sigma\_{t-1}^{p}(I+A\\Delta t)^{T}+Q\\Delta t$$
+Rearranging this terms will give
+
+$$\\frac{1}{\\Delta t}(\\Sigma_t^p-\\Sigma\_{t-1}^p) = -\\frac{k\_{t-1}C\\Sigma\_{t-1}^p}{\\Delta t} + (A\\Sigma\_{t-1}^p+AK\_{k-1}C\\Sigma\_{t-1}^p+\\Sigma\_{t-1}^pA^T-K\_{t-1}C\\Sigma\_{t-1}^pA^T+Q) + O(\\Delta t^2)$$
+
+With limit, $Δt → 0$
+
+We get the finite difference version,
+$$\\underbrace{\\dot{\\Sigma}(t)=A\\Sigma(t)+\\Sigma(t)A^{T}-\\Sigma(t)C^{T}R^{-1}C\\Sigma(t)+Q}\_{\\text{Riccati ODE}}$$
+
+### Duality in the Riccati Equation
+
```

Comment:
A brief written description of what you mean would be helpful.  Something about how you are drawing parallels between the process of LQR and development of a CKF.

Reply:
Added.